File: benchmarks-QuTiP/qfunc_state.py
import qutip as qt
import numpy as np
import benchmarkutils

name = "qfunc_state"

samples = 3
evals = 5
cutoffs = range(10, 101, 10)

def setup(N):
    alpha = 0.7
    xvec = np.linspace(-50, 50, 100)
    yvec = np.linspace(-50, 50, 100)
    state = qt.coherent(N, alpha)
    return state, xvec, yvec

def f(state, xvec, yvec):
    return qt.qfunc(state, xvec, yvec)

print("Benchmarking:", name)
print("Cutoff: ", end="", flush=True)
checks = {}
results = []
for N in cutoffs:
    print(N, "", end="", flush=True)
    state, xvec, yvec = setup(N)
    alpha_check = 0.6 + 0.1j
    checks[N] = f(state, [alpha_check.real], [alpha_check.imag])[0, 0]
    t = benchmarkutils.run_benchmark(f, state, xvec, yvec, samples=samples, evals=evals)
    results.append({"N": N, "t": t})
print()

benchmarkutils.check(name, checks)
benchmarkutils.save(name, results)

File: benchmarks-QuantumOptics.jl/qfunc_state.jl
using QuantumOptics
using BenchmarkTools
include("benchmarkutils.jl")

name = "qfunc_state"

samples = 3
evals = 5
cutoffs = [10:10:100;]

function setup(N)
    alpha = 0.7
    xvec = collect(linspace(-50, 50, 100))
    yvec = collect(linspace(-50, 50, 100))
    b = FockBasis(N-1)
    state = coherentstate(b, alpha)
    state, xvec, yvec
end

function f(state, xvec, yvec)
    qfunc(state, xvec, yvec)
end

println("Benchmarking: ", name)
print("Cutoff: ")
checks = Dict{Int, Float64}()
results = []
for N in cutoffs
    print(N, " ")
    state, xvec, yvec = setup(N)
    alpha_check = 0.6 + 0.1im
    checks[N] = f(state, [real(alpha_check)], [imag(alpha_check)])[1, 1]
    t = @belapsed f($state, $xvec, $yvec) samples=samples evals=evals
    push!(results, Dict("N"=>N, "t"=>t))
end
println()

benchmarkutils.check(name, checks)
benchmarkutils.save(name, results)
